When I try to compile your configuration I get this error:

```console
CP:  /home/linguini/coding/nuttx-space/nuttx/include/nuttx/config.h
CC:  grp/lib_getgrbuf.c generator.c: In function ‘replay_worker’:
generator.c:205:44: error: ‘const struct orb_metadata’ has no member named ‘o_format’
  205 |                        sensor_gen->obj.meta->o_format, data);
      |                                            ^~
generator.c: In function ‘fake_worker’:
generator.c:287:38: error: ‘const struct orb_metadata’ has no member named ‘o_format’
  287 |                  sensor_gen->obj.meta->o_format, data) < 0)
      |                                      ^~
make[2]: *** [/home/linguini/coding/nuttx-space/apps/Application.mk:330: generator.c.home.linguini.coding.nuttx-space.apps.system.uorb.o] Error 1
make[2]: *** Waiting for unfinished jobs....
IN: mm/libmm.a -> staging/libmm.a make[1]: *** [Makefile:54: /home/linguini/coding/nuttx-space/apps/system/uorb_all] Error 2
make[1]: *** Waiting for unfinished jobs....
CC:  pthread/pthread_mutexattr_setpshared.c make: *** [tools/LibTargets.mk:248: /home/linguini/coding/nuttx-space/apps/libapps.a] Error 2
make: *** Waiting for unfinished jobs....
```

--- a/system/uorb/listener.c
+++ b/system/uorb/listener.c
@@ -330,24 +330,25 @@ static int listener_update(FAR struct listen_list_s *objlist,
 
       delta_time       = now_time - old->timestamp;
       delta_generation = state.generation - old->generation;
+
+      unsigned long frequency = 0;
       if (delta_generation && delta_time)
         {
-          unsigned long frequency;
-
           frequency = (state.max_frequency ? state.max_frequency : 1000000)
                       * delta_generation / delta_time;
-          uorbinfo_raw("\033[K" "%-*s %2u %4" PRIu32 " %4lu "
-                       "%2" PRIu32 " %4u",
-                       ORB_MAX_PRINT_NAME,
-                       object->meta->o_name,
-                       object->instance,
-                       state.nsubscribers,
-                       frequency,
-                       state.queue_size,
-                       object->meta->o_size);
-          old->generation = state.generation;
-          old->timestamp  = now_time;
         }
+
+      uorbinfo_raw("\033[K" "%-*s %2u %4" PRIu32 " %4lu "
+                   "%2" PRIu32 " %4u",
+                   ORB_MAX_PRINT_NAME,
+                   object->meta->o_name,
+                   object->instance,
+                   state.nsubscribers,
+                   frequency,
+                   state.queue_size,
+                   object->meta->o_size);
+      old->generation = state.generation;
+      old->timestamp  = now_time;
     }
   else
     {

